File: spring-boot-rest-api/src/main/java/my/company/springbootrestapi/model/Place.java
package my.company.springbootrestapi.model;

import javax.persistence.*;
import java.util.List;

@Entity
public class Place {

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long id;

    private String name;
    private String address;

    @OneToMany(cascade = {CascadeType.ALL})
    private List<Period> periods;

    public Place() {
    }

    public Place(String name, String address, List<Period> period) {
        this.name = name;
        this.address = address;
        this.periods = period;
    }

    public Place(Place place) {
        this.name = place.name;
        this.address = place.address;
        this.periods = place.periods;
    }


    public List<Period> getPeriods() {
        return periods;
    }

    public Long getId() {
        return id;
    }

    public String getName() {
        return name;
    }

    public String getAddress() {
        return address;
    }


}

File: angular-front-end/src/app/model/place.ts
import {Period} from "./period";
import {IPlace} from "./iplace";
import {IPeriod} from "./iperiod";

export class Place implements IPlace {
    id: number;
    address: string;
    name: string;
    periods: Period[];

    constructor(place: IPlace) {
        this.id = place.id;
        this.name = place.name;
        this.address = place.address;
        this.periods = place.periods.map(period => new Period(period));
    }

}
